dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -103,21 +106,42 @@ DataInterfaceIIDM::build(std::string iidmFilePath) {
 #endif
     boost::split(paths, extensionsPaths, boost::is_any_of(splitCharacter));
 
-    for (unsigned int i = 0; i < paths.size(); ++i) {
-      std::regex fileRegex(stdcxx::format(".*libiidm-ext-.*\\%1%", boost::dll::shared_library::suffix().string()));
-      powsybl::iidm::ExtensionProviders<powsybl::iidm::converter::xml::ExtensionXmlSerializer>::getInstance().loadExtensions(paths[i], fileRegex);
+    {
+      std::unique_lock<std::mutex> lock(loadExtensionMutex_);
+      for (unsigned int i = 0; i < paths.size(); ++i) {
+        std::regex fileRegex(stdcxx::format(".*libiidm-ext-.*\\%1%", boost::dll::shared_library::suffix().string()));
+        powsybl::iidm::ExtensionProviders<powsybl::iidm::converter::xml::ExtensionXmlSerializer>::getInstance().loadExtensions(paths[i], fileRegex);
```

Comment:
What happens if several threads execute this code?

Reply:
mutex will hanle this

Reply:
no, mutex will make sure they are not called at the same time by 2 threads, are we sure powsybl handle properly that this method is called multiple times?